participant/app/utils/validation.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from "zod";
+import { zfd } from "zod-form-data";
+
+const nameSchemaFactory = (idPrefix: string) => {
+  /* eslint-disable-rule @typescript-eslint/no-unused-vars
+     I do not understand why eslint thinks we're not using these vars
+  */
+  const firstNameKey = `${idPrefix}-firstName`;
+  const lastNameKey = `${idPrefix}-lastName`;
+  const preferredNameKey = `${idPrefix}-preferredName`;
+  return zfd.formData({
+    [firstNameKey]: zfd.text(
+      z.string().min(1, { message: "First name is required" })
+    ),
+    [lastNameKey]: zfd.text(
+      z.string().min(1, { message: "Last name is required" })
```

Comment:
Should these `message`s be shown on the form? Should these `message`s be i18n strings?

<img width="544" alt="Screen Shot 2023-03-27 at 1 01 02 PM" src="https://user-images.githubusercontent.com/67701/228054013-5de29faf-f4a8-420d-bb00-f12df2abbfec.png">

Reply:
Nevermind! This should be part of https://wicmtdp.atlassian.net/browse/PRP-202 so I retract my comment